### PR TITLE
Transfer all item shadow functions to the item class

### PR DESCRIPTION
**Pull request by annando**
The file include/threads.php is now ready to be transferred to src/Model/Thread.php since it now only contains thread table related functions.

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -60,4 +62,143 @@ public static function update(array $fields, array $condition)
...
+		}
+
+		// Only add a shadow, if the profile isn't hidden
+		if (dba::exists('user', ['uid' => $item['uid'], 'hidewall' => true])) {
+			return;
+		}
+
+		$item = dba::selectFirst('item', [], ['id' => $itemid]);
+
+		if (count($item) && ($item["allow_cid"] == '')  && ($item["allow_gid"] == '') &&
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please use `DBM::is_result` instead of `count()`

File: src/Model/Item.php [hunk 2]
```diff
@@ -60,4 +62,143 @@ public static function update(array $fields, array $condition)
...
+			return;
+		}
+
+		// Is there already a shadow entry?
+		if (dba::exists('item', ['uri' => $item['uri'], 'uid' => 0])) {
+			return;
+		}
+
+		// Preparing public shadow (removing user specific data)
+		require_once("include/items.php");
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please remove parentheses.

File: src/Worker/ThreadUpdate.php [hunk 3]
```diff
@@ -1,11 +0,0 @@
-<?php
-namespace Friendica\Worker;
-
-require_once("include/threads.php");
-
-class ThreadUpdate {
```
Comment (MrPetovan) [inline src/Worker/ThreadUpdate.php]: Isn’t this mentioned anywhere?

Comment (annando) [reply src/Worker/ThreadUpdate.php]: We don't need that anymore. It was part of old update routines that we removed.

File: src/Model/Item.php (hunk 1 above)
Comment (annando) [reply src/Model/Item.php]: Sorry, I forgot this.

**Pull request merged**